libraries/chain/include/eosio/chain/authority.hpp

```diff
@@ -7,15 +7,15 @@
 
 namespace eosio { namespace chain {
 
-using shared_public_key_data = fc::static_variant<fc::ecc::public_key_shim, fc::crypto::r1::public_key_shim, shared_string>;
+using shared_public_key_data = std::variant<fc::ecc::public_key_shim, fc::crypto::r1::public_key_shim, shared_string>;
 
 struct shared_public_key {
    shared_public_key( shared_public_key_data&& p ) :
       pubkey(std::move(p)) {}
 
    operator public_key_type() const {
       fc::crypto::public_key::storage_type public_key_storage;
-      pubkey.visit(overloaded {
```

Comment:
visit has similar issue, static_variant::visit throws  FC_THROW_EXCEPTION( fc::assert_exception, "Internal error: static_variant tag is invalid.");  std:visit throws std::bad_variant_access, need similar wrap as fc::get